src/components/map/tools/ExtendRoutePathTool.ts

```diff
@@ -0,0 +1,62 @@
+import RoutePathStore from '~/stores/routePathStore';
+import NodeType from '~/enums/nodeType';
+import ToolbarTool from '~/enums/toolbarTool';
+import ErrorStore from '~/stores/errorStore';
+import { INode } from '~/models';
+import RoutePathLinkService from '~/services/routePathLinkService';
+import BaseTool from './BaseTool';
+
+/**
+ * Tool for creating new routePath
+ */
+class ExtendRoutePathTool implements BaseTool {
+    public toolType = ToolbarTool.AddNewRoutePathLink;
+    public activate() {}
+    public deactivate() {
+        this.reset();
+    }
+
+    private reset() {
+        RoutePathStore.setNeighborRoutePathLinks([]);
+    }
+
+    public onNetworkNodeClick = async (clickEvent: any) => {
+        try {
+            if (!this.isNetworkNodesInteractive()) return;
+
+            const properties =  clickEvent.sourceTarget.properties;
```

Comment:
nip: there are two spaces between = and clickEvent.

Reply:
Nit* (nitpicking)